### DISCUSSION TITLE
Fail baking tegra-uefi-capsules.bb for Orin AGX (do_compile needs to run as root)

### DISCUSSION THREAD

**Discussion by jpeeee** [OE4T/meta-tegra#1677] [Q&A]
Hello,

I am using the meta-tegra layer at 24503b40b6671cffcb5806ec68972ef77e5a798c (p3737-0000-p3701-0005: set correct TEGRA_FLASHVARs, KERNEL_DEVICETREE and EMMC_BCT, Signed-off-by: Timo Schuster).

I am getting the error below when trying to bitbake tegra-uefi-capsules.

The warning in the error log seems to indicate one of the scripts expects to run as root: "Please run this script as root or using sudo!". 

Manually running the run.do_compile script as root (sudo  .../tegra-uefi-capsules/36.3.0/temp/run.do_compile) 

I started looking at fakeroot (never explicitly used it before), thinking it could be used to run the do_compile as root, but I was not able to run the do_compile stage as root using it. 
do_compile() {
    do_compile_task
}
fakeroot do_compile_task() {
    # Generate BUP images
   ...
}

Any help and/or pointers to resolve this issue appreciated.

DEBUG: Executing python function extend_recipe_sysroot
NOTE: Direct dependencies are ['/workdir/layers/meta-tegra/recipes-bsp/tegra-binaries/tegra-bootfiles_36.3.0.bb:do_populate_sysroot', '/workdir/layers/meta-tegra/recipes-bsp/tegra-binaries/tegra-flashtools-native_36.3.0.bb:do_populate_sysroot', '/workdir/layers/meta-tegra/recipes-bsp/uefi/edk2-basetools-tegra-native_36.3.0.bb:do_populate_sysroot', '/workdir/layers/meta-tegra/recipes-kernel/nvidia-kernel-oot/nvidia-kernel-oot_36.3.0.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-core/glibc/cross-localedef-native_2.39.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-core/glibc/glibc_2.39.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-core/glibc/ldconfig-native_2.12.1.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-devtools/gcc/gcc-cross_13.2.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-devtools/gcc/gcc-runtime_13.2.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-devtools/qemu/qemuwrapper-cross_1.0.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-devtools/quilt/quilt-native_0.67.bb:do_populate_sysroot', '/workdir/layers/meta/recipes-kernel/kmod/depmodwrapper-cross_1.0.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-core/coreutils/coreutils_9.4.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-core/update-rc.d/update-rc.d_0.8.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/apt/apt_2.6.1.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/dosfstools/dosfstools_4.2.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/dpkg/dpkg_1.22.0.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/makedevs/makedevs_1.0.1.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/mtools/mtools_4.0.43.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/opkg-utils/opkg-utils_0.6.3.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/patch/patch_2.7.6.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/perl/perl_5.38.2.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/pseudo/pseudo_git.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/python/python3-pyyaml_6.0.1.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-devtools/python/python3_3.12.3.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-extended/cpio/cpio_2.15.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-extended/pigz/pigz_2.8.bb:do_populate_sysroot', 'virtual:native:/workdir/layers/meta/recipes-kernel/dtc/dtc_1.7.0.bb:do_populate_sysroot']
NOTE: Installed into sysroot: []
NOTE: Skipping as already exists in sysroot: ['tegra-bootfiles', 'tegra-eks-image-base', 'tegra-eks-image', 'tegra-flashtools-native', 'tegra-flashvars', 'tegra-helper-scripts-native', 'tegra-storage-layout-base', 'tegra-storage-layout', 'edk2-basetools-tegra-native', 'nvidia-kernel-oot', 'openssl', 'expat', 'gettext-minimal-native', 'cross-localedef-native', 'glibc', 'ldconfig-native', 'libxcrypt', 'ncurses', 'systemd-systemctl-native', 'util-linux-libuuid', 'zlib', 'binutils-cross-aarch64', 'cmake-native', 'gcc-cross-aarch64', 'gcc-runtime', 'libgcc', 'libedit', 'libtool-native', 'm4-native', 'opkg-utils', 'python3', 'qemu-native', 'qemuwrapper-cross', 'quilt-native', 'bzip2', 'libnsl2', 'libtirpc', 'texinfo-dummy-native', 'xz', 'depmodwrapper-cross', 'linux-libc-headers', 'gdbm', 'libffi', 'sqlite3', 'openssl-native', 'coreutils-native', 'expat-native', 'gettext-native', 'glib-2.0-native', 'ncurses-native', 'update-rc.d-native', 'util-linux-libuuid-native', 'util-linux-native', 'zlib-native', 'apt-native', 'dosfstools-native', 'dpkg-native', 'flex-native', 'gnu-config-native', 'libedit-native', 'make-native', 'makedevs-native', 'mtools-native', 'opkg-utils-native', 'patch-native', 'perl-native', 'pseudo-native', 'python3-build-native', 'python3-cython-native', 'python3-flit-core-native', 'python3-installer-native', 'python3-packaging-native', 'python3-pyproject-hooks-native', 'python3-pyyaml-native', 'python3-setuptools-native', 'python3-wheel-native', 'python3-native', 'bzip2-native', 'cpio-native', 'libidn2-native', 'libnsl2-native', 'libtirpc-native', 'pigz-native', 'shadow-native', 'unzip-native', 'xz-native', 'zstd-native', 'dtc-native', 'kmod-native', 'attr-native', 'db-native', 'debianutils-native', 'dos2unix-native', 'gdbm-native', 'gmp-native', 'gnutls-native', 'libtasn1-native', 'libbsd-native', 'libcap-ng-native', 'libcap-native', 'libffi-native', 'libgcrypt-native', 'libgpg-error-native', 'libmd-native', 'libmpc-native', 'libpcre2-native', 'libunistring-native', 'libyaml-native', 'lz4-native', 'mpfr-native', 'nettle-native', 'sqlite3-native', 'xxhash-native']
DEBUG: Python function extend_recipe_sysroot finished
DEBUG: Executing shell function do_compile
NOTE: Copying EXTERNAL_KERNEL_DEVICETREE entry tegra234-p3737-0000+p3701-0005-nv.dtb to .
NOTE: Copying overlay L4TConfiguration.dtbo to .
NOTE: Copying overlay L4TConfiguration-RootfsRedundancyLevelABEnable.dtbo to .
NOTE: Copying external overlay tegra234-p3737-0000+p3701-0000-dynamic.dtbo to .
NOTE: Copying external overlay tegra234-carveouts.dtbo to .
NOTE: Copying external overlay tegra-optee.dtbo to .
NOTE: Copying external overlay tegra234-p3737-camera-dual-imx274-overlay.dtbo to .
NOTE: Copying external overlay tegra234-p3737-camera-e3331-overlay.dtbo to .
NOTE: Copying external overlay tegra234-p3737-camera-e3333-overlay.dtbo to .
NOTE: Copying external overlay tegra234-p3737-camera-imx185-overlay.dtbo to .
NOTE: Copying overlay L4TConfiguration-rcmboot.dtbo to .
Please run this script as root or using sudo!
Please run this script as root or using sudo!
mv: cannot stat '/workdir/build/tmp/work/horizon_orin_devkit_64gb-mdt-linux/tegra-uefi-capsules/36.3.0/tegra-uefi-capsules-36.3.0/bup-payload/payloads_t23x/*': No such file or directory
WARNING: /workdir/build/tmp/work/horizon_orin_devkit_64gb-mdt-linux/tegra-uefi-capsules/36.3.0/temp/run.do_compile.15505:170 exit 1 from 'mv /workdir/build/tmp/work/horizon_orin_devkit_64gb-mdt-linux/tegra-uefi-capsules/36.3.0/tegra-uefi-capsules-36.3.0/bup-payload/payloads_t23x/* .'
WARNING: Backtrace (BB generated script): 
	#1: do_compile_task, /workdir/build/tmp/work/horizon_orin_devkit_64gb-mdt-linux/tegra-uefi-capsules/36.3.0/temp/run.do_compile.15505, line 170
	#2: do_compile, /workdir/build/tmp/work/horizon_orin_devkit_64gb-mdt-linux/tegra-uefi-capsules/36.3.0/temp/run.do_compile.15505, line 146
	#3: main, /workdir/build/tmp/work/horizon_orin_devkit_64gb-mdt-linux/tegra-uefi-capsules/36.3.0/temp/run.do_compile.15505, line 431

**Comment by ichergui** [comment]
Hi @jpeeee 
Could you please join our element channel and we can discuss about your issue ? 
https://matrix.to/#/#nv-jetson-secureboot:gitter.im

**Comment by jpeeee** [comment]
@ichergui done!

**Comment by ichergui** [comment]
The issue was fixed. so, I'm closing this discussion.
